backend/service/github/github_test.go

```diff
@@ -672,3 +677,110 @@ func TestGetRepository(t *testing.T) {
 		})
 	}
 }
+
+type mockPullRequests struct {
+	generalError bool
+
+	actualNumber     int
+	actualHTMLURL    string
+	actualBranchName string
+}
+
+// Dummy mock of Create API so mockPullRequests implements v3pullrequests
+func (m *mockPullRequests) Create(ctx context.Context, owner string, repo string, pull *githubv3.NewPullRequest) (*githubv3.PullRequest, *githubv3.Response, error) {
+	return &githubv3.PullRequest{}, &githubv3.Response{}, nil
+}
+
+// Mock of ListPullRequestsWithCommit API
+func (m *mockPullRequests) ListPullRequestsWithCommit(ctx context.Context, owner, repo, sha string, opts *githubv3.PullRequestListOptions) ([]*githubv3.PullRequest, *githubv3.Response, error) {
+	if m.generalError {
+		return nil, nil, errors.New(problem)
+	}
+
+	m.actualNumber = 1347
+	m.actualHTMLURL = fmt.Sprintf("https://github.com/%s/%s/pull/%s", owner, repo, strconv.Itoa(m.actualNumber))
+	m.actualBranchName = "my-branch"
+
+	return []*githubv3.PullRequest{
+		{
+			Number:  intPtr(m.actualNumber),
+			State:   strPtr("open"),
```

Comment:
should this be the `opts.State`?

Reply:
def a nit, just to make the mock more representative of what they passed in

Reply:
Hmm I believe it should just be State. This is a mock of https://github.com/lyft/clutch/blob/4bd65211d507baf6d2c95a6fc73ca8a0c74e9446/backend/service/github/iface.go#L37 which returns `[]*githubv3.PullRequest` where `PullRequest` is defined as https://github.com/google/go-github/blob/1d9884e7ac8ca3c6d3d469af28af227fc773a246/github/pulls.go#L30

Reply:
You can think of this mock as the mock for what the Github API returns in https://github.com/lyft/clutch/blob/b4651bc598b000cfbf7dea033b3151b4c7d06180/backend/service/github/github.go#L246

Reply:
sorry, I meant this:
```suggestion
			State:   strPtr(opts.State),
```

Reply:
I guess what I'm saying is this mock is hardcoded to return an "open" PR as opposed to something that the user can control, which is what the other args seem to do

Reply:
ah I see what you mean, good catch. updated